cpp/CMakeLists.txt

```diff
@@ -14,19 +14,57 @@
 # limitations under the License.
 #=============================================================================
 
-cmake_minimum_required(VERSION 3.12...3.17 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.18 FATAL_ERROR)
+
+# If `CMAKE_CUDA_ARCHITECTURES` is not defined, build for all supported architectures. If
+# `CMAKE_CUDA_ARCHITECTURES` is set to an empty string (""), build for only the current
+# architecture. If `CMAKE_CUDA_ARCHITECTURES` is specified by the user, use user setting.
+
+# This needs to be run before enabling the CUDA language due to the default initialization behavior
+# of `CMAKE_CUDA_ARCHITECTURES`, https://gitlab.kitware.com/cmake/cmake/-/issues/21302
+if(NOT DEFINED CMAKE_CUDA_ARCHITECTURES OR CMAKE_CUDA_ARCHITECTURES STREQUAL "ALL")
+  set(CUSPATIAL_BUILD_FOR_ALL_ARCHS TRUE)
+elseif(CMAKE_CUDA_ARCHITECTURES STREQUAL "")
+  set(CUSPATIAL_BUILD_FOR_DETECTED_ARCHS TRUE)
+endif()
 
 project(CUDA_SPATIAL VERSION 0.20.0 LANGUAGES C CXX CUDA)
```

Comment:
I think this is a bad merge as it's renamed in 0.19: https://github.com/rapidsai/cuspatial/blob/branch-0.19/cpp/CMakeLists.txt#L31

Reply:
Yeah, I guess I picked the wrong one.